modules/post/linux/recon/host_info.rb

```diff
@@ -0,0 +1,109 @@
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::Linux::System
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Update information about session host',
+        'Description' => %q{
+          Collects as much basic information on the current session as possible.
+          Metherpreter sessions may do (most) of these functions automatically,
+          but this module allows gathering even on basic shells in case a
+          meterpreter session isnt viable.
+          This is meant to help recon more data on a target or correct data assumed
+          from external scans.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Nick Cottrell <ncottrellweb[at]gmail.com>'],
+        'Platform' => ['linux', 'unix', 'bsd'],
+        'Privileged' => false,
+        'SessionTypes' => %w[meterpreter shell],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
```

Comment:
Since running this module doesn't result in a new session, let's remove this based on the info from here: https://docs.metasploit.com/docs/development/developing-modules/module-metadata/definition-of-module-reliability-side-effects-and-stability.html#:~:text=the%20first%20attempt-,REPEATABLE_SESSION,-The%20module%20is
```suggestion
          'Reliability' => [],
```

Reply:
So it kind of can. It can be repeated if you change the kernel name, but I cannot get the random name generation to happen without the help of a function, which I can do but I'm not sure if I should or if there's a better way to do it.